helpers/config.py

```diff
@@ -616,28 +616,36 @@ def validate_passwords(self):
             },
         }
         errors = []
+        psql_replication = passwords.pop('PostgreSQL replication')
         for label, config_ in passwords.items():
             if not re.match(config_['pattern'], config_['password']):
                 errors.append(label)
                 CLI.colored_print(
                     f'{label} password contains unsupported characters.',
-                    CLI.COLOR_WARNING
+                    CLI.COLOR_ERROR
                 )
         if errors:
-            CLI.colored_print(
-                'You should run `python run.py --setup` to update.',
-                CLI.COLOR_ERROR
-            )
-            # PostgreSQL replication password is set in PostgreSQL on the first
-            # launch and nothing is run afterwards in subsequent starts to update
-            # it if it has changed.
-            if 'PostgreSQL replication' in errors:
                 CLI.colored_print(
-                    'PostgreSQL replication password must be changed manually\n'
-                    'in `kobo-install/.run.conf` and PostgreSQL itself.',
+                    'You should run `python run.py --setup` to update.',
                     CLI.COLOR_WARNING
                 )
 
+        # PostgreSQL replication password must be handled separately because
+        # it is set in PostgreSQL on the first launch and nothing is done
+        # afterwards in subsequent starts to update it if it has changed.
+        if not re.match(
+            psql_replication['pattern'], psql_replication['password']
+        ):
+            CLI.colored_print(
+                'PostgreSQL replication password contains unsupported characters.',
+                CLI.COLOR_ERROR
+            )
```

Comment:
@noliveleger, this is maybe a stupid question, but if the unsupported characters issue comes from https://github.com/kobotoolbox/kpi/pull/3709, is the replication password even affected? The Django apps never read it, right—only Postgres itself does?

Reply:
OK, thanks.
:telephone_receiver: :eyes: :pensive: